### PR TITLE
#170947433 Implement user registration

### PR DESCRIPTION
**Pull request by AbdoulNuru**
#### What does this PR do?

- Implement user registration feature
- Use a 3rd API to send account verification emails.

#### How should this be manually tested?

- clone this repo
- install the dependencies by running `npm i` in your terminal.
- by referencing .env.example create a .env file and add all the keys.
- run `npm run migrateDb` in your terminal to run the migrations
- run `npm run dev` in your terminal to start the server.
- use postman to test this route `/api/v1/auth/signup` you will have to provide the following fields: firstName, lastName, email, password.
- check your email inbox, you will find an email from barefoot nomad.

#### Implementation plan

[User registration](https://docs.google.com/document/d/1qByH2hXJoWXIr9jFQIi3Co-gRLSjU_amzJQ3PnHsHDk/edit)

#### Stories associated with PR

[#170947433](https://www.pivotaltracker.com/story/show/170947433)

#### Screenshots if any
Screen Shot 2020-02-11 at 13 31 22
Screen Shot 2020-02-11 at 13 32 28

### CODE REVIEW

File: src/database/seeders/20200206095927-dummy-user.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-    lastName: 'Octapus',
-    email: 'octapus@andela.com',
+    userID: 'd0a051d9-447a-49a8-aebc-7e1b031afd62',
+    firstName: 'Octopus',
+    lastName: 'Octopus',
+    email: 'octopus@andela.com',
+    password: '123',
+    isVerified: false,
     createdAt: new Date(),
     updatedAt: new Date()
```
Comment (blaise82) [inline src/database/seeders/20200206095927-dummy-user.js]: is it possible that you may add a `method` too?

**Review by izzett222 — APPROVED** [review]

**Review by blaise82 — CHANGES_REQUESTED** [review]

File: src/database/seeders/20200206095927-dummy-user.js (hunk 1 above)
Comment (blaise82) [reply src/database/seeders/20200206095927-dummy-user.js]: is it possible that you may add a hashed password 
for **'password: '123''** won't have any use

File: src/tests/signup.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import Chai from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../index';
+
+Chai.use(chaiHttp);
+Chai.should();
+
+const user = {
```
Comment (hakoemmy) [inline src/tests/signup.test.js]: @AbdoulNuru I think you can put this mock data in the mock folder
Put this 
```suggestion
const user = {
  firstName: 'Nuru',
  lastName: 'Abd',
  email: 'abdoulniyigena@gmail.com',
  password: 'nuru1234'
};
```
in file
   ``
tests/mockData/user.mock.js `` like that.

**Review by hakoemmy — CHANGES_REQUESTED** [review]

**Review by RusimbiPatrick — APPROVED** [review]

File: src/database/seeders/20200206095927-dummy-user.js (hunk 1 above)
Comment (AbdoulNuru) [reply src/database/seeders/20200206095927-dummy-user.js]: @blaise82 Thank you, it is done

File: src/tests/signup.test.js (hunk 2 above)
Comment (AbdoulNuru) [reply src/tests/signup.test.js]: @hakoemmy Thank you, it is done

**Review by blaise82 — APPROVED** [review]

**Review by hakoemmy — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -4,24 +4,28 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "start": "node build/index.js",
+    "start": "node --require @babel/polyfill build/index.js",
     "dev": "nodemon --exec babel-node src/index.js",
-    "test": "nyc --reporter=html --reporter=text mocha  --require @babel/register 'src/**/**test.js' --exit",
+    "test": "NODE_ENV=test npm run migrateTest && NODE_ENV=test nyc --reporter=html --reporter=text mocha  --require @babel/polyfill --require @babel/register 'src/**/**test.js' --exit",
```
Comment (kcharles52) [inline package.json]:
```suggestion
    "test": "NODE_ENV=test npm run migrateTest && nyc --reporter=html --reporter=text mocha  --require @babel/polyfill --require @babel/register 'src/**/**test.js' --exit",
```

File: src/controllers/user.controller.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+     * @param {object} res
+     * @returns {object} newUser
+     * @memberof userController
+     */
+  static async signUp(req, res) {
+    try {
+      const {
+        firstName, lastName, email, password
+      } = req.body;
+
```
Comment (kcharles52) [inline src/controllers/user.controller.js]:
```suggestion
```

File: src/controllers/user.controller.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      const newUser = await models.Users.create({
+        userID: uuid(), method: 'local', firstName, lastName, email, password, isVerified: false
+      });
+
```
Comment (kcharles52) [inline src/controllers/user.controller.js]:
```suggestion
```

remove these spaces

File: src/routes/api/user.routes.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ *                 type: string
+ *               password:
+ *                 type: string
+ *     produces:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               status:
+ *                 type: string
```
Comment (kcharles52) [inline src/routes/api/user.routes.js]: Please check the status type

**Review by kcharles52 — CHANGES_REQUESTED** [review]

**Review by kcharles52 — APPROVED** [review]

**Pull request merged**